admin/category_product_listing.php

```diff
@@ -870,26 +870,32 @@
             );
             break;
           case 'delete_category':
+            $childs_count = zen_childs_in_category_count($cInfo->categories_id);
+            $products_count = zen_products_in_category_count($cInfo->categories_id);
+            $has_linked = false; 
+            $prod_list = $db->Execute("SELECT * FROM " . TABLE_PRODUCTS_TO_CATEGORIES . " WHERE categories_id = " . $cInfo->categories_id); 
+            foreach ($prod_list as $prod) {
+               $other_cat_list = $db->Execute("SELECT * FROM " . TABLE_PRODUCTS_TO_CATEGORIES . " WHERE products_id = " . $prod['products_id'] . " AND  categories_id != " . $prod['categories_id']);
+               if (!$other_cat_list->EOF) {
+                  $has_linked = true; 
+                  break; 
+               }
+            }
```

Comment:
Isn't there a function which does this foreach loop already?

Reply:
It's for a different category.

Reply:
I was thinking of something like this, so that the queries are kept within existing functions that are already doing these lookups:
```suggestion
            $childs_count = zen_childs_in_category_count($cInfo->categories_id);
            $products_count = zen_products_in_category_count($cInfo->categories_id);
            $has_linked = false; 
            $prod_list = zen_get_linked_products_for_category($cInfo->categories_id);
            foreach ($prod_list as $prod) {
               if (zen_get_linked_categories_for_product($prod, $cInfo->categories_id)) {
                  $has_linked = true; 
                  break; 
               }
            }
```

Reply:
OK I see.  I misread your earlier comment - I was thinking you were referring to code higher up in the file (which does indeed do a count but for a different category).  I'll get this done shortly.

Reply:
Your code doesn't quite work - second parm to zen_get_linked_categories_for_product() is supposed to be an array.  But the problem is this also uncovers a bug in zen_get_linked_categories_for_product().  I'm going to put both in this update, please split if you wish.

Reply:
Good catch. Thanks.